core/network/impl/gossiper_broadcast.cpp

```diff
@@ -5,87 +5,42 @@
 
 #include "network/impl/gossiper_broadcast.hpp"
 
+#include <atomic>
+#include <memory>
+
 #include "network/common.hpp"
-#include "network/helpers/scale_message_read_writer.hpp"
 #include "network/impl/loopback_stream.hpp"
 
 namespace kagome::network {
+  KAGOME_DEFINE_CACHE(stream_engine);
 
   GossiperBroadcast::GossiperBroadcast(libp2p::Host &host)
-      : host_{host}, logger_{common::createLogger("GossiperBroadcast")} {}
+      : logger_{common::createLogger("GossiperBroadcast")},
+        stream_engine_{StreamEngine::create(host)} {}
```

Comment:
I'm not sure that creating a stream engine here avoiding DI will have a good influence on testabilty.